ui/ui-guidelines.md

```diff
@@ -71,13 +100,134 @@ are reflected in how the codebase is organized.
+- Enum members and constants are snakecase `ALL_CAPS`.
+- Other variables are `camelCase`.
+- Non-component directories are `kebab-case`.
+
+## React components
+
+### Component directory structure
+
+- In library packages that contain more than just components, components should live in a top-level `components` directory.
+- React components should always live in a directory with a matching name and an index.ts file that exports the contents of the directory that we want to expose for public consumption.
+- The component directory may include additional components that help with composing the component (e.g. a `List` component may also include a `ListItem` component, a complex component may be broken down into pieces for ease of code maintenance).
+- The component directory should also include closely associated files like unit tests (`*.test.tsx`) and stories (`*.stories.tsx`).
+
+### Component responsibilities and composition
+
+- When designing components and their APIs for libraries, assume they may be used individually, in a dashboard, and alongside other components that are not dashboards.
+- It is better to err on the side of building more components with less responsibilities than a single component with a lot of responsibilities. This makes it easier to use components as composable building blocks in a wide variety of use cases.
+- If a component has multiple boolean flag props, consider to split it into multiple
+  components instead of making it do too many things.
+- If you find yourself doing significant data transformation or business logic inside a component, think about if it may want to be used outside that component. If so, consider pulling it out into a utility function or wrapper component that is more reusable.
+- When possible, avoid building atomic components (e.g. buttons, alerts, dropdowns) from scratch. Instead use or build on top of MUI components.
+
+### Component props
+
+- All components with props should define their types in a prop using the naming scheme `ComponentNameProps` and export that type for external use.
+- Be thoughtful about exposing props from MUI when using it as a tool to build components. Do we actually want consumers to be able to override all of those values? Will we be able to support them doing so?
+
+### Styling components
+
+- Most of our CSS lives inline with components because we believe that the look
+  and behavior of a component should be coupled
+- Use the [`SxProps` API in Material UI](https://mui.com/system/getting-started/the-sx-prop/) for doing most custom styling inline with
+  the components. We occasionally use the `styled` hook.
+- Material UI is themeable, so we define any global overrides for Material UI
+  components in the theme
+- The theme includes our color palette, typography, spacing units and more and is
+  available and used by our custom component styles as well
+- When writing CSS, *always* use the variables from our Theme instead of writing
```

Comment:
I would put the word `always` in bold instead and in uppercase to insist on this rule.